### PR TITLE
Start of the index page

### PR DESCRIPTION
**Pull request by orta**
Ports over just the first article in the home page, includes a lot of unused assets and a few unused components which I will need as I get further but this is fine to ship as-is
image

### CODE REVIEW

**Comment by orta** [comment]
OK, preview seems have issues with my use of a global stylesheet 👍🏻

File: docs/public/homepage.scss [hunk 1]
```diff
@@ -0,0 +1,303 @@
+@font-face {
```
Comment (antfu) [inline docs/public/homepage.scss:1]: You can move this file to `.vitepress/theme` and import it in `.vitepress/theme/index.ts`, it's the main JS entry

Comment (orta) [reply docs/public/homepage.scss]: Perfect, that did it - good to go

File: docs/components/home/Homepage.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <br>
+            and ex-TypeScript compiler team member
+          </p>
+
+          <p class="by" style="margin-top: 1rem">
+            And <a href="https://orta.io">Anthony Fu</a>
+          </p>
+
+          <p>
+            A veritable open sourceror.
```
Comment (antfu) [inline docs/components/home/Homepage.vue:43]: Haha, thank you! 😊

File: docs/components/home/Homepage.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            By <a href="https://orta.io">orta therox</a>
+          </p>
+          <p>
+            Purveyor of <a href="https://puzzmo.com/">Puzzles</a>, shipper of renowned open source code
+            <br>
+            and ex-TypeScript compiler team member
+          </p>
+
+          <p class="by" style="margin-top: 1rem">
+            And <a href="https://orta.io">Anthony Fu</a>
```
Comment (antfu) [inline docs/components/home/Homepage.vue:39]:
```suggestion
            And <a href="https://antfu.me">Anthony Fu</a>
```

File: docs/components/home/Homepage.vue [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  <main class="main">
+    <article class="container border-red">
+      <div class="intro">
+        <p>
+          The documentation concerning
+          <br>
+          the npm modules of
+        </p>
+        <h1 class="title">
+          Shiki-Twoslash
```
Comment (antfu) [inline docs/components/home/Homepage.vue:14]:
```suggestion
          Twoslash
```

File: docs/components/home/Homepage.vue [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            And <a href="https://orta.io">Anthony Fu</a>
+          </p>
+
+          <p>
+            A veritable open sourceror.
+          </p>
+        </div>
+
+        <div style="margin-top: 3rem">
+          <img style="margin: 0 auto" src="/svgs/logo.svg" alt="Shiki Logo" width="167" height="238">
```
Comment (antfu) [inline docs/components/home/Homepage.vue:48]:
```suggestion
          <img style="margin: 0 auto" src="/logo.svg" alt="Shiki Logo" width="167" height="238">
```

For the logo, I made a modified version by adding a beige background so it works better on dark backgrounds. Not sure what you would prefer:

image

Source is here btw: https://www.figma.com/file/sCQBSGPqSASpiql84BZ8Ct/TwoSlash?type=design&node-id=0%3A1&mode=design&t=CaY0IU8Xjz9NIBvo-1

File: docs/components/home/Homepage.vue [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        <p>
+          In which markdown code samples are powered by
+          <br>
+          the syntax engine of Visual Studio Code
+          <br>
+          mixed with the TypeScript compiler’s information
+        </p>
+
+        <div style="margin-top: 3rem; margin-bottom:3rem">
+          <img style="margin: 0 auto" src="/svgs/squiggle.svg" alt="Decoration" width="70" height="25.5">
```
Comment (antfu) [inline docs/components/home/Homepage.vue:25]:
```suggestion
          <img class="dark:invert" style="margin: 0 auto" src="/svgs/squiggle.svg" alt="Decoration" width="70" height="25.5">
```

Little trick to support dark mode :P

**Comment by antfu** [comment]
Feel free to merge :)

**Comment by orta** [comment]
The rest of the homepage does not handle dark mode, will need to handle that later

**Pull request merged**